float64 param
---
bool success
